packages/grid_client/src/zos/network_light.ts

```diff
@@ -0,0 +1,31 @@
+import { Expose, Type } from "class-transformer";
+import { IsNotEmpty, IsOptional, IsString, ValidateNested } from "class-validator";
+
+import { ValidateMembers } from "../helpers";
+import { WorkloadData } from "./workload_base";
+
+class Mycelium {
+  @Expose() @IsString() @IsNotEmpty() hex_key: string;
+  @Expose() @IsOptional() @IsString({ each: true }) peers?: string[];
+}
+
+@ValidateMembers()
+class NetworkLight extends WorkloadData {
+  @Expose() @IsString() @IsNotEmpty() subnet: string;
+  @Expose() @IsOptional() @Type(() => Mycelium) @ValidateNested() mycelium?: Mycelium;
+
+  challenge(): string {
+    let out = "";
+    out += this.subnet;
+
+    out += this.mycelium?.hex_key || "";
+    if (this.mycelium?.peers) {
+      for (let i = 0; i < this.mycelium?.peers?.length; i++) {
+        out += this.mycelium?.peers[i] || "";
+      }
+    }
+    return out;
```

Comment:
```suggestion
 ```
   let out = this.subnet || ""; 
out += this.mycelium?.hex_key || "";   
if (this.mycelium?.peers) {  
  out += this.mycelium.peers.map(peer => peer || "").join(""); 
}  
return out;
```
```

Reply:
I will dismiss this as it's just coding style and I'm following the other zos classes as it might affect signatures and challenges